forem/forem
# Remove Liquid::BlockBody patch

## What type of PR is this? (check all applicable)


## Description

Currently most tests that use the model are broken on `master`, the reason being a not working patch to `Liquid::BlockBody` added to https://github.com/forem/forem/pull/12150 to disable the [newly introduced](https://github.com/Shopify/liquid/pull/1086) `{% liquid %}` liquid tag (yep). Unfortunately that tag wasn't implemented as a regular tag thus the monkeypatch introduced seemed to work at the moment but had side-effects that broke our code. 

The thing that's still unclear to me is how https://github.com/forem/forem/commit/a63ba17b6a4d131ca18015ba8effc2146136962e managed to be deployed if all tests were broken (the Travis build is green, both the buildbot's and the merge commit's!) but if you run `bundle exec rspec spec/models/article_spec.rb` locally, for example, you'll see most tests fail because of the patch.

A lingering question remains: should we make an effort to disable that rendering tag? `echo`, see https://github.com/Shopify/liquid/pull/1086, is disabled, but `liquid` is not. 

You can test it with `Liquid::Template.parse("{% liquid %}")` on this branch

## Related Tickets & Documents

- https://github.com/forem/forem/pull/12150
- https://github.com/Shopify/liquid/pull/1086

## QA Instructions, Screenshots, Recordings

1. on `master` run `bundle exec rspec spec/models/article_spec.rb` and see most tests fail
2. do the same on this branch and see the specs pass

## Added tests?

- [x] No, and this is why: removed the code that was breaking tests
